### PR TITLE
redirect validation resource

### PR DESCRIPTION
**Pull request by wesleymccollam**

### CODE REVIEW

**Comment by henryrecker-pingidentity** [comment]
Do we have the HCL formatting linter in this repo? The spacing of the HCL example looks a little off

File: internal/provider/provider.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	oauthAuthServerSettingsScopesExclusiveScopes "github.com/pingidentity/terraform-provider-pingfederate/internal/resource/oauthauthserversettingsscopesexclusivescopes"
 	oauthIssuer "github.com/pingidentity/terraform-provider-pingfederate/internal/resource/oauthissuers"
 	protocolMetadataLifetimeSettings "github.com/pingidentity/terraform-provider-pingfederate/internal/resource/protocolmetadatalifetimesettings"
+	redirectValidation "github.com/pingidentity/terraform-provider-pingfederate/internal/resource/redirectvalidation"
```
Comment (henryrecker-pingidentity) [inline internal/provider/provider.go:26]: We probably don't need to rename these imports

File: internal/types/client_utils.go [hunk 2]
```diff
@@ -37,6 +37,74 @@ func ToStateAuthenticationPolicyContract(r *client.AuthenticationPolicyContract)
 	return caSliceOfObj, eaSliceOfObj
 }
 
+func AttrValueToBoolPointer(val attr.Value) *bool {
+	boolPointerVal := ConvertToPrimitive(val).(bool)
```
Comment (henryrecker-pingidentity) [inline internal/types/client_utils.go:41]: Probably better to do this by converting to a types.Bool and then using the ValueBoolPointer method

File: internal/resource/redirectvalidation/redirect_validation_resource.go [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readRedirectValidationResponse(ctx context.Context, r *client.RedirectValidationSettings, state *redirectValidationResourceModel, expectedValues *redirectValidationResourceModel) {
+	state.Id = types.StringValue("id")
+	stateRedirectValidationLocalSettings, _ := internaltypes.ToStateRedirectValidation(r, diag.Diagnostics{})
+	state.RedirectValidationLocalSettings = stateRedirectValidationLocalSettings
+	_, stateRedirectValidationPartnerSettings := internaltypes.ToStateRedirectValidation(r, diag.Diagnostics{})
```
Comment (henryrecker-pingidentity) [inline internal/resource/redirectvalidation/redirect_validation_resource.go:273]: You could combine 271 and 273 into one call so you don't run the method twice

File: internal/resource/redirectvalidation/redirect_validation_resource.go [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readRedirectValidationResponse(ctx context.Context, r *client.RedirectValidationSettings, state *redirectValidationResourceModel, expectedValues *redirectValidationResourceModel) {
```
Comment (henryrecker-pingidentity) [inline internal/resource/redirectvalidation/redirect_validation_resource.go:269]: Looks like expectedValues isn't needed in this method

File: internal/types/client_utils.go [hunk 5]
```diff
@@ -37,6 +37,74 @@ func ToStateAuthenticationPolicyContract(r *client.AuthenticationPolicyContract)
...
+	return &boolPointerVal
+}
+
+func ToStateRedirectValidation(r *client.RedirectValidationSettings, diags diag.Diagnostics) (basetypes.ObjectValue, basetypes.ObjectValue) {
```
Comment (henryrecker-pingidentity) [inline internal/types/client_utils.go:45]: If this only applies to this new resource I would move it to the resource Go file

File: internal/resource/redirectvalidation/redirect_validation_resource.go (hunk 4 above)
Comment (wesleymccollam) [reply internal/resource/redirectvalidation/redirect_validation_resource.go]: Fixed in the latest commit

File: internal/types/client_utils.go (hunk 5 above)
Comment (wesleymccollam) [reply internal/types/client_utils.go]: Fixed in the latest commit

File: internal/resource/redirectvalidation/redirect_validation_resource.go (hunk 3 above)
Comment (wesleymccollam) [reply internal/resource/redirectvalidation/redirect_validation_resource.go]: This was moved to the resource file in the latest commit.

File: internal/types/client_utils.go (hunk 2 above)
Comment (wesleymccollam) [reply internal/types/client_utils.go]: Removed this completely as it's not used.

File: internal/provider/provider.go (hunk 1 above)
Comment (wesleymccollam) [reply internal/provider/provider.go]: Imports were fixed as a part of another branch recently merged into main. These have been merged and the new structure is applied in the latest commit for this resource.

File: internal/acctest/acctest.go [hunk 6]
```diff
@@ -153,6 +153,13 @@ func TestAttributesMatchBool(resourceType string, resourceName *string, attribut
 	return nil
 }
 
+func TestAttributesMatchBoolPointer(resourceType string, resourceName *string, attributeName string, expected, found *bool) error {
+	if expected != found {
+		return mismatchedAttributeError(resourceType, resourceName, attributeName, strconv.FormatBool(*expected), strconv.FormatBool(*found))
```
[the remark this answers is not in the document]
Comment (wesleymccollam) [reply internal/acctest/acctest.go]: Removed this completely as it's not used.

**Review by henryrecker-pingidentity — APPROVED** [review]

**Pull request merged**